Common/ComponentSystemGuiCoreNg/Object/BasePanel/BasePanelDockWidget.cpp

```diff
@@ -0,0 +1,297 @@
+#include "BasePanelDockWidget.h"
+#include <QtCore/QDebug>
+
+#define TIMER_WAIT_MOVE         0.5 // время ожидания до показа/скрытия
+#define MOVE_VALUE              100 // размер в px увеличения/уменьшения размера за 1 итерацию
+#define MOUSE_AREA_ACTIVATE     20  // область в px при попадании в которую покажется панель (если атоматически скрывается)
+
+BasePanelDockWidget::BasePanelDockWidget(const QString &title, QWidget *parent)
+    : QDockWidget(title, parent),
+      _mdiArea(nullptr)
+{
+    _state = MoveVisible;
+    _isAutoHide = false;
+    _isPanelVisible = true;
+
+    _timerWaitMove.setInterval (TIMER_WAIT_MOVE * 1000);
+    connect (&_timerWaitMove, SIGNAL(timeout()), this, SLOT(waitTimeOut()));
```

Comment:
`connect(&_timerWaitMove, &QTimer::timeout, this, &BasePanelDockWidge::waitTimeOut);`?